pxr/usd/usdShade/validators.cpp

```diff
@@ -192,13 +192,48 @@ _ShaderPropertyTypeConformance(const UsdPrim &usdPrim)
     return errors;
 }
 
+static bool _HasMaterialBindingRelationship(const UsdPrim& usdPrim) {
```

Comment:
Might be better to make this a lambda within `_MaterialBindingApiAppliedValidator` method?

And also make the `materialBindingString` a `static const`, or its totally fine if you do `UsdShadeTokens->materialBinding.GetString()` in place.